### PR TITLE
Feat/field type user property

### PR DESCRIPTION
**Pull request by Davidmattei**
|Q              |A  |
|---------------|---|
|Bug fix?       |n|
|New feature?   |y|	
|BC breaks?     |n|	
|Deprecations?  |n|	
|Fixed tickets  |n|	

Add new fieldType for selecting a user property (displayName, userName or email).
Allow the user to add a different value. (allow_add option on fieldType)
Support multiple selections
Add role limitation for only showing users that have at least 1 of the selected roles.

### CODE REVIEW

File: src/Form/DataField/SelectUserPropertyFieldType.php [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    /**
+     * @param FormBuilderInterface<FormBuilderInterface> $builder
+     * @param array<mixed>                               $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        parent::buildForm($builder, $options);
+
+        /** @var FieldType $fieldType */
+        $fieldType = $options['metadata'];
```
Comment (theus77) [inline src/Form/DataField/SelectUserPropertyFieldType.php:58]: Test instansceof?

File: src/Form/DataField/SelectUserPropertyFieldType.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    public function getBlockPrefix(): string
+    {
+        return 'bypassdatafield';
+    }
+
+    /**
+     * @param FormBuilderInterface<FormBuilderInterface> $builder
+     * @param array<mixed>                               $options
```
Comment (theus77) [inline src/Form/DataField/SelectUserPropertyFieldType.php:51]:
```suggestion
     * @param array<string, mixed>                               $options
```

File: src/Form/DataField/SelectUserPropertyFieldType.php [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+                'allow_add' => true,
+                'multiple' => false,
+                'user_roles' => [],
+            ])
+        ;
+    }
+
+    /**
+     * @param FormBuilderInterface<FormBuilderInterface> $builder
+     * @param array<mixed>                               $options
```
Comment (theus77) [inline src/Form/DataField/SelectUserPropertyFieldType.php:85]:
```suggestion
     * @param array<string, mixed>                               $options
```

**Review by theus77 — APPROVED** [review]
Nice trick to add choices via events

File: src/Form/DataField/SelectUserPropertyFieldType.php (hunk 3 above)
Comment (Davidmattei) [reply src/Form/DataField/SelectUserPropertyFieldType.php]: 16db8f8

File: src/Form/DataField/SelectUserPropertyFieldType.php (hunk 2 above)
Comment (Davidmattei) [reply src/Form/DataField/SelectUserPropertyFieldType.php]: 16db8f8

File: src/Form/DataField/SelectUserPropertyFieldType.php (hunk 1 above)
Comment (Davidmattei) [reply src/Form/DataField/SelectUserPropertyFieldType.php:57]: 16db8f8

**Review by dameert — APPROVED** [review]

**Pull request merged**